crates/mun_runtime/benches/fibonacci.rs

```diff
@@ -0,0 +1,102 @@
+use criterion::{black_box, criterion_group, criterion_main, BenchmarkId, Criterion};
+use mun_runtime::invoke_fn;
+
+mod util;
+
+/// A benchmark test that runs fibonacci(n) for a number of samples and compares the performance
+/// for calling the implementation between different languages.
+pub fn fibonacci_benchmark(c: &mut Criterion) {
+    // Perform setup (not part of the benchmark)
+    let runtime = util::runtime_from_file("fibonacci.mun");
+    let lua = util::lua_from_file("fibonacci.lua");
+    let wasm = util::wasmer_from_file("fibonacci.wasm");
+
+    let mut group = c.benchmark_group("fibonacci");
+
+    // Iterate over a number of samples
+    for i in [100i64, 200i64, 500i64, 1000i64, 4000i64, 8000i64].iter() {
+        // Run Mun fibonacci
+        group.bench_with_input(BenchmarkId::new("mun", i), i, |b, i| {
+            b.iter(|| {
+                let _: i64 = invoke_fn!(runtime, "main", *i).unwrap();
+            })
+        });
+
+        // Run Rust fibonacci
+        group.bench_with_input(BenchmarkId::new("rust", i), i, |b, i| {
+            b.iter(|| fibonacci_main(*i))
+        });
+
+        // Run LuaJIT fibonacci
+        group.bench_with_input(BenchmarkId::new("luajit", i), i, |b, i| {
+            b.iter(|| {
+                let f: mlua::Function = lua.globals().get("main").unwrap();
+                let _: i64 = f.call(*i).unwrap();
+            })
+        });
+
+        // Run Wasm fibonacci
+        group.bench_with_input(BenchmarkId::new("wasm", i), i, |b, i| {
+            b.iter(|| {
+                wasm.call("main", &[(*i as i32).into()]).unwrap();
+            })
+        });
+    }
+
+    group.finish();
+
+    fn fibonacci(n: i64) -> i64 {
```

Comment:
Why are there both a fibonacci.rs and these functions?

Reply:
Good catch, I removed the files in resources. I want to include the benchmark function in the benchmark itself to show what a reference implementation looks like.

Reply:
What is the source material for the *.wasm modules and simple instruction on how to generate them for new tests?

Reply:
I took the basic example: https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_wasm. But instead of `greet` I implemented the functionality. But that could ofcourse be done with any language.